lexer.js

```diff
@@ -268,13 +268,11 @@ function commentWhitespace () {
 
 function templateString () {
   while (charCode = str.charCodeAt(++i)) {
-    if (charCode === 36/*$*/) {
-      charCode = str.charCodeAt(++i);
-      if (charCode === 123/*{*/) {
-        templateStack.push(templateDepth);
-        templateDepth = ++braceDepth;
-        return;
-      }
+    if (charCode === 36/*$*/ && str.charCodeAt(i + 1) === 123/*{*/) {
+      charCode = str.charCodeAt(i += 2);
+      templateStack.push(templateDepth);
+      templateDepth = ++braceDepth;
+      return;
     }
     else if (charCode === 96/*`*/) {
```

Comment:
Thanks for the PR!

There's a slightly easier fix - to just change this to an `if` instead of an `else if`!

I've added that and the test case from this PR to the Wasm branch, hoping to release that tomorrow.